jerry-debugger/jerry_client_ws.py

```diff
@@ -683,13 +342,162 @@ def __init__(self, address):
+        self._exec_command(JERRY_DEBUGGER_MEMSTATS)
+
+    def set_break(self, args):
+        _set_breakpoint(self, args, False)
+        if self.breakpoint_info != '':
+            sbp = self.breakpoint_info
+            self.breakpoint_info = ''
+            return sbp
+        return None
+
+    def delete(self, args):
+        result = ''
+        if args == "all":
+            self.delete_active()
+            self.delete_pending()
+        elif args == "pending":
+            self.delete_pending()
+        elif args == "active":
+            self.delete_active()
+        else:
+            try:
+                breakpoint_index = int(args)
+            except ValueError as val_errno:
+                result = "Error: Integer number expected, %s" % (val_errno)
+                return result
+
+            if breakpoint_index in self.active_breakpoint_list:
+                breakpoint = self.active_breakpoint_list[breakpoint_index]
+                del self.active_breakpoint_list[breakpoint_index]
+                breakpoint.active_index = -1
+                self.send_breakpoint(breakpoint)
+            elif breakpoint_index in self.pending_breakpoint_list:
+                del self.pending_breakpoint_list[breakpoint_index]
+                if not self.pending_breakpoint_list:
+                    self.send_parser_config(0)
+            else:
+                result = "Error: Breakpoint %d not found" % (breakpoint_index)
+        return result
+
+    def backtrace(self, args):
+        max_depth = 0
+
+        if args:
+            try:
+                max_depth = int(args)
+                if max_depth <= 0:
+                    self.smessage += "Error: Positive integer number expected"
+            except ValueError as val_errno:
+                self.smessage += "Error: Positive integer number expected, %s" % (val_errno)
+
+        message = struct.pack(self.byte_order + "BBIB" + self.idx_format,
+                              WEBSOCKET_BINARY_FRAME | WEBSOCKET_FIN_BIT,
+                              WEBSOCKET_FIN_BIT + 1 + 4,
+                              0,
+                              JERRY_DEBUGGER_GET_BACKTRACE,
+                              max_depth)
+        self.send_message(message)
+
+    def eval(self, code):
+        self._send_string(JERRY_DEBUGGER_EVAL_EVAL + code, JERRY_DEBUGGER_EVAL)
+
+    def throw(self, code):
+        self._send_string(JERRY_DEBUGGER_EVAL_THROW + code, JERRY_DEBUGGER_EVAL)
+
+    def abort(self, args):
+        self.delete("all")
+        self.exception("0")  # disable the exception handler
+        self._send_string(JERRY_DEBUGGER_EVAL_ABORT + args, JERRY_DEBUGGER_EVAL)
+
+    def restart(self):
+        self._send_string(JERRY_DEBUGGER_EVAL_ABORT + "\"r353t\"", JERRY_DEBUGGER_EVAL)
+
+    def exception(self, args):
+        enable = 0
+        result = ''
+        if not args:
+            result = "Error: Status expected!"
+        else:
+            enable = int(args)
```

Comment:
This will leads to a crash if I gave a non integer argument. Please handle it also.